### PR TITLE
#979 Update Outbound line mutations - This changes EVERYTHING -- [11]

### LINKED ISSUE #979: Sending the right line edit mutations
For each of the line edit saving mutations the logic needs to be adjusted:

Outbound:
- Only send mutations for packs with zero or more packs. Delete if updating to zero.

Inbound: 
- Only send mutations for packs with zero or more packs. Delete if updating to zero.

Stocktake:
- Should only insert or update if `countedNumberOfPacks` is not `null`
- Should delete if `countedNumberOfPacks` is `null`

Response
- I think nothing. Supplying zero quantity is legitimate

Request
- I think nothing. We can send updates for zero quantity with this. Also when we add from master list we have lines with requested quantity of zero

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #979 Part1 😁 

- This updates the outbound line mutation so that when setting a lines number of packs to zero, the line is deleted. Otherwise, the update is sent and the API throws and error
- I was also working on inbound and refactored a little bit too much, so thought to do another PR with a different diff

### CODE REVIEW

File: packages/invoices/src/OutboundShipment/api/api.ts [hunk 1]
```diff
@@ -237,16 +237,30 @@ export const getOutboundQueries = (sdk: Sdk, storeId: string) => ({
     const input = {
       insertOutboundShipmentLines: draftStocktakeLines
         .filter(
-          ({ type, isCreated }) =>
-            isCreated && type === InvoiceLineNodeType.StockOut
+          ({ type, isCreated, numberOfPacks }) =>
+            isCreated &&
+            type === InvoiceLineNodeType.StockOut &&
+            numberOfPacks > 0
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/api.ts:243]: Not the most elegant code! suggestions welcome 🤔

File: packages/invoices/src/OutboundShipment/api/api.ts [hunk 2]
```diff
@@ -276,12 +293,10 @@ export const getOutboundQueries = (sdk: Sdk, storeId: string) => ({
 
     return result;
   },
-  deleteLines: (invoiceId: string) => async (ids: string[]) => {
+  deleteLines: async (lines: { id: string; invoiceId: string }[]) => {
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/api.ts:296]: I wanted to consolidate upsert/delete to use an array of line objects so changed the param on this

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 3]
```diff
@@ -269,27 +267,19 @@ export const useSaveOutboundLines = () => {
...
-export const useDeleteInboundLine = (): UseMutationResult<
-  DeleteOutboundShipmentLinesMutation,
-  unknown,
-  string[],
-  { previous?: OutboundFragment; ids: string[] }
-> => {
-  // TODO: Shouldn't need to get the invoice ID here from the params as the mutation
-  // input object should not require the invoice ID. Waiting for an API change.
-  const { data } = useOutbound();
+export const useDeleteInboundLines = () => {
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:270]: which made me look at this and then I started.. tidying?

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 4]
```diff
@@ -317,33 +311,19 @@ export const useDeleteSelectedLines = (): {
...
     } else {
-      return {
-        selectedRows: (
-          Object.keys(state.rowState)
-            .filter(id => state.rowState[id]?.isSelected)
-            .map(selectedId => lines.find(({ id }) => selectedId === id))
-            .filter(Boolean) as OutboundLineFragment[]
-        ).map(({ id }) => id),
-      };
+      return lines.filter(({ id }) => state.rowState[id]?.isSelected);
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:326]: this seems a little nicer!

File: packages/invoices/src/OutboundShipment/api/api.ts (hunk 1 above)
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/api/api.ts:243]: uh, when testing an earlier PR and having problems.. I did exactly this! 😆

File: packages/invoices/src/OutboundShipment/api/hooks.ts (hunk 4 above)
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/api/hooks.ts:326]: much 😄 
Might almost be small enough to 

```
    return isGrouped
      ? items
          ?.filter(({ id }) => state.rowState[id]?.isSelected)
          .map(({ lines }) => lines.flat())
          .flat()
      : lines.filter(({ id }) => state.rowState[id]?.isSelected);
```

**Review by mark-prins — APPROVED** [review]
🎉  it does change everything!!

File: packages/invoices/src/OutboundShipment/api/hooks.ts (hunk 4 above)
Comment (joshxg) [reply packages/invoices/src/OutboundShipment/api/hooks.ts:326]: Nice! I see the suggestion in another PR also, will do both at once

**Pull request merged**